lib/screens/map_view.dart

```diff
@@ -52,64 +52,75 @@ class _MapViewScreenState extends State<MapViewScreen> {
       }
 
       locationData = await locationService.getLocation();
-
     } on PlatformException catch (e) {
       debugPrint('Error: ${e.toString()}, code: ${e.code}');
       locationData = null;
     }
-    locationData = await locationService.getLocation();
     setState(() {});
   }
 
   Widget map(BuildContext context) {
+    MapControllerImpl mapController = MapControllerImpl();
     if (locationData == null) {
       return const Center(child: CircularProgressIndicator());
     } else {
-      return FlutterMap(
-        options: MapOptions(
-          center: latlng.LatLng(
-              locationData?.latitude ?? 51.5, locationData?.longitude ?? -0.09),
-          zoom: 18.0,
-          interactiveFlags: InteractiveFlag.none,
-        ),
-        layers: [
-          TileLayerOptions(
-              urlTemplate: mapboxURL, // ignore: undefined_identifier
-              attributionBuilder: (_) {
-                return const Text("© Mapbox");
-              },
-              additionalOptions: {
-                'accessToken': mapboxAPIKey, // ignore: undefined_identifier
-                'id': 'mapbox.mapbox-streets-v8'
-              }),
-          MarkerLayerOptions(
-            markers: [
-              Marker(
-                  width: 40.0,
-                  height: 40.0,
-                  point: latlng.LatLng(locationData?.latitude ?? 51.5,
-                      locationData?.longitude ?? -0.09),
-                  builder: (ctx) => const Beastie())
-            ],
-          ),
-        ],
-      );
+      return StreamBuilder(
+          stream: locationService.onLocationChanged.asBroadcastStream(),
+          builder: (context, AsyncSnapshot<LocationData> currLocation) {
+            if (!currLocation.hasData) {
+              return const Center(child: CircularProgressIndicator());
+            }
+            latlng.LatLng mapCenter = latlng.LatLng(
+                currLocation.data?.latitude ?? 51.5,
+                currLocation.data?.longitude ?? -0.09);
+            const zoomLevel = 18.0;
+            mapController.onReady.then((_) {
```

Comment:
I think this little bit was probably the most important part. The MapController basically lets you interact with the map and if we didn't explicitly tell it to move, it would stay frozen in place while our marker left on a journey